benchmarks/functional_autograd_benchmark/audio_text_models.py

```diff
@@ -0,0 +1,128 @@
+import torch
+import torch.nn as nn
+
+import torchaudio_models as models
+
+from utils import make_functional, load_weights
+
+def get_wav2letter(device):
+    N = 10
+    input_frames = 700
+    vocab_size = 28
+    model = models.Wav2Letter(num_classes=vocab_size)
+    criterion = torch.nn.NLLLoss()
+    model.to(device)
+    params, names = make_functional(model)
+
+    inputs = torch.rand([N, 1, input_frames], device=device)
+    labels = torch.rand(N, 3, device=device).mul(vocab_size).long()
+
+    def forward(*new_params):
+        load_weights(model, names, new_params)
```

Comment:
One thing that surprised me about the code is that `make_functional` modifies `model` in-place (it removes all parameters from `model`). This was surprising because of two reasons:
- I thought that something with the word `functional` in its name (`make_functional`) would be functional and not modify any of its inputs.
- In `load_weights(model, names, new_params)`, we're still using `model` 

Is it possible to have `make_functional` return `(params, names, model_without_weights)`, or to rename model to `model_without_weights` or something?
```
params, names = make_functional(model)
model_without_weights = model  # NB: make_functional removed weights from `model`.
```

Reply:
We can't really "clone" the model easily. So it is indeed modified inplace.
I can update the namings to make it clearer.

And yes functional is a strong word here. We can actually have side effect. We just make sure we get get the Parameters as input instead of just capturing them.